cli-config/src/config.rs

```diff
@@ -64,6 +64,11 @@ pub struct Config {
     /// `solana_sdk::commitment_config::CommitmentLevel::Confirmed`.
     #[serde(default)]
     pub commitment: String,
+    /// A moniker override config
+    ///
+    /// Optional YAML map to override moniker settings to custom set ones
+    #[serde(default)]
+    pub moniker_override: Option<HashMap<String, String>>,
```

Comment:
I think it's time for a CLI-specific `enum Cluster` (for the key here).  Ala https://github.com/solana-labs/solana/blob/4bc3a1195fbdb2ec31f48f82d28e644ce881f6aa/sdk/src/genesis_config.rs#L44-L50 but with the values matching (and replacing!) https://github.com/solana-labs/solana/blob/4bc3a1195fbdb2ec31f48f82d28e644ce881f6aa/clap-utils/src/input_validators.rs#L214-L222 This can live in its own module in `clap-utils`.

We can also ditch the `Option` and just `#[serde(skip_serializing_if = "HashMap::is_empty")]`.  The query can then look like `self.moniker_override.get(custom_url).unwrap_or(public_url)`

Reply:
right on. I can look into this later tonight